--- /usr/share/lipstick-jolla-home-qt5/lockscreen/Lockscreen.qml.1.1.4	2015-04-22 23:52:05.133769751 +0200
+++ /usr/share/lipstick-jolla-home-qt5/lockscreen/Lockscreen.qml	2015-04-22 23:50:17.783390380 +0200
@@ -13,6 +13,7 @@
 import org.nemomobile.configuration 1.0
 import "../notifications"
 import "../scripts/desktop.js" as Desktop
+import MeeGo.QOfono 0.2
 
 MouseArea {
     id: lockScreen
@@ -153,6 +154,17 @@
                 FadeAnimation { }
             }
         }
+        /* Voicemail notification */
+        Image {
+            x: Theme.paddingLarge
+            anchors {
+                left: parent.left
+                leftMargin: 33
+            }
+            source: "/usr/share/themes/jolla-ambient/meegotouch/z1.0/icons/icon-s-voicemail.png"
+            visible: ofonoMessageWaiting.voicemailWaiting
+        }
+
     }
 
     SimToolkitIdleModeIndicator {
@@ -253,6 +265,15 @@
         anchors.bottom: clock.top
     }
 
+    /* Voicemail notification */
+    OfonoManager { id: ofonoManager }
+
+    OfonoMessageWaiting {
+        id: ofonoMessageWaiting
+        modemPath: ofonoManager.modems.length > 0 ? ofonoManager.modems[0] : ""
+    }
+
+
     SneakPeekHint {
         id: sneakPeekHint
     }

